### PR TITLE
3 inventory items sprint 2

### PR DESCRIPTION
**Pull request by mitsudz**
2nd Team 3 pr to main

- Refactoring of Hotbar code
- Integrating player stats with item usage - items now when clicked will actually have an effect on the players stats (e.g. hunger etc.)

Closes #238 

@PHONGNGYUEN1 

@Pratulw5 
@mitsudz

### CODE REVIEW

File: source/core/assets/images/foodtextures/download.png
Comment (mitsudz) [inline source/core/assets/images/foodtextures/download.png:1]: This removes a duplicate file that made it to main.

File: source/core/assets/configs/ForestGameAreaConfig.json
Comment (mitsudz) [inline source/core/assets/configs/ForestGameAreaConfig.json:1]: This removes a duplicate config class which made it to main and shouldn't be used.

File: source/core/src/main/com/csse3200/game/components/CombatStatsComponent.java [hunk 1]
```diff
@@ -22,8 +22,8 @@ public class CombatStatsComponent extends Component {
   private final int maxExperience;
 
   public CombatStatsComponent(int health, int hunger, int strength, int defense, int speed, int experience) {
-      this.maxHealth = health;
-      this.maxHunger = hunger;
+      this.maxHealth = 100;
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/components/CombatStatsComponent.java:25]: Allowing for health to be initialised to something other than the input health (probably should add this as an input to the constructor.

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java [hunk 2]
```diff
@@ -1,137 +1,111 @@
...
-             // ensure it's added to the stage
-            toggleHotbar = true;
-        }
-    }
-
-
+    /**
+     * Toggles the visibility of the hotbar.
+     * If the hotbar is currently visible, it will be removed, and vice versa.
+     */
```
Comment (owenrharding) [inline source/core/src/main/com/csse3200/game/components/player/PlayerInventoryHotbarDisplay.java:55]: Removed function?

**Review by owenrharding — APPROVED** [review]
Looks fine to me

**Review by LucasRyall — APPROVED** [review]
Looks good, a lot of magic numbers, especially with the potions, still approving but would like to see this changed at some point (makes it easier to work on the code in future).

File: source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java [hunk 3]
```diff
@@ -82,16 +79,17 @@ public void create() {
      * Toggles the inventory display on or off based on its current state.
      */
     private void toggleInventory() {
-        hotbar.toggleHotbar();
         if (stage.getActors().contains(window, true)) {
             logger.debug("Inventory toggled off.");
```
[the remark this answers is not in the document]
Comment (mitsudz) [reply source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java:83]: They are in the correct order - if the stage already contains the inventory window it must have been toggled on in the first place.

**Pull request merged**